packages/fuels-code-gen/src/program_bindings/abigen/configurables.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::HashSet;
+
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+
+use crate::{
+    error::Result,
+    program_bindings::{
+        abi_types::{FullConfigurable, FullTypeDeclaration},
+        resolved_type::{resolve_type, ResolvedType},
+    },
+    utils::safe_ident,
+};
+
+#[derive(Debug)]
+pub(crate) struct ResolvedConfigurable {
+    pub name: Ident,
```

Comment:
You can, if you like, use `Component` here as it also represents a field/variant in a struct/enum. It will resolve the type and snake-case the name.

Reply:
We decided to not modify the string so I will leave the current implementation